auth.go

```diff
@@ -51,16 +51,27 @@ type dockerConfig struct {
 func NewAuthConfigurationsFromDockerCfg() (*AuthConfigurations, error) {
```

Comment:
This function is conflating finding the correct path to use, with opening said path and creating the NewAuthConfiguration struct. I would split this logic for finding the path out into FindConfigPath or the like and then this function becomes simpler and more single purpose in nature. Testing might feel a bit less icky also as you can write the appropriate tests around FindConfigPath and have them sprawl a bit less as well as only require two test around the NewAuthConfigurationsFromDockerCfg (one fail on path one works on path).